frontend/src/app/Settings/Facility/FacilityForm.tsx

```diff
@@ -226,16 +180,20 @@ const FacilityForm: React.FC<Props> = (props) => {
   };
 
   const validateFacilityAddresses = async () => {
-    const originalFacilityAddress = getFacilityAddress(facility);
-
+    let updatedValues = getUpdatedValues();
+    const originalFacilityAddress = getFacilityAddress(updatedValues.facility);
     const zipCodeData = await getZipCodeData(originalFacilityAddress.zipCode);
     const isValidZipForState = isValidZipCodeForState(
       originalFacilityAddress.state,
       zipCodeData
     );
 
     if (!isValidZipForState) {
-      showError("Invalid ZIP code for this state", "Form Errors");
+      setError("facility.zipCode", {
+        type: "validZipForState",
+        message: "Invalid ZIP code for this state",
```

Comment:
similar note to the comment [here](https://github.com/CDCgov/prime-simplereport/pull/5983/files#r1235287974)